src/server/modes/charts/plugins/request-with-dataset/request-dataset.ts

```diff
@@ -81,15 +81,47 @@ export const getDatasetFields = async (args: {
     let datasetFields: PartialDatasetField[];
     let revisionId: string;
 
-    const cacheResponse = await cacheClient.get({key: cacheKey});
-
-    if (cacheResponse.status === Cache.OK) {
-        datasetFields = cacheResponse.data.datasetFields;
-        revisionId = cacheResponse.data.revisionId;
-        req.ctx.log('DATASET_FIELDS_WAS_RECEIVED_FROM_CACHE');
+    if (cacheClient.client) {
+        const cacheResponse = await cacheClient.get({key: cacheKey});
+
+        if (cacheResponse.status === Cache.OK) {
+            datasetFields = cacheResponse.data.datasetFields;
+            revisionId = cacheResponse.data.revisionId;
+            req.ctx.log('DATASET_FIELDS_WAS_RECEIVED_FROM_CACHE');
+        } else {
+            req.ctx.log('DATASET_FIELDS_IN_CACHE_WAS_NOT_FOUND');
+
+            const response = await getDatasetFieldsById(
+                datasetId,
+                req,
+                rejectFetchingSource,
+                iamToken,
+                pluginOptions,
+            );
+            datasetFields = response.fields;
+            revisionId = response.revision_id;
+            cacheClient
+                .set({
+                    key: cacheKey,
+                    ttl: pluginOptions?.cache || DEFAULT_CACHE_TTL,
+                    value: {datasetFields, revisionId},
+                })
+                .then((setCacheResponse) => {
+                    if (setCacheResponse.status === Cache.OK) {
+                        req.ctx.log('SET_DATASET_IN_CACHE_SUCCESS');
+                    } else {
+                        req.ctx.logError('SET_DATASET_FIELDS_IN_CACHE_FAILED', setCacheResponse);
+                        req.ctx.logError(
```

Comment:
Is it necessary to log error twice?

Reply:
@jhoncool Oh, just an artifact. Removed it in e3f4431986